src/basis/entity.js

```diff
@@ -321,121 +323,126 @@
     };
   };
 
-  var ENTITYSET_INIT_METHOD = function(superClass, name){
-    return function(){
+
+ /**
+  * @class
+  */
+  var EntitySet = Class(Dataset, {
+    className: namespace + '.EntitySet',
+
+    name: null,
+    wrapper: $self,
+
+    init: function(){
       if (!this.name)
-        this.name = getUntitledName(name);
+        this.name = getUntitledName('EntitySet');
 
-      // inherit
-      superClass.prototype.init.call(this);
-    };
-  };
+      Dataset.prototype.init.call(this);
+    },
+    destroy: function(){
+      Dataset.prototype.destroy.call(this);
 
-  var ENTITYSET_SYNC_METHOD = function(){
-    return function(data){
+      this.name = null;
+      this.wrapper = null;
+    },
+
+    add: ENTITYSET_WRAP_METHOD(Dataset, 'add'),
+    remove: ENTITYSET_WRAP_METHOD(Dataset, 'remove'),
+    set: ENTITYSET_WRAP_METHOD(Dataset, 'set'),
+
+    setAndDestroyRemoved: function(items){
       if (this.localId)
-        return this.set(data);
+        return this.set(items);
```

Comment:
`data` name was ok, as it may be not an array of models (but could be), but raw objects also